lib/Canonicalize/MakeModelGEPPass.cpp

```diff
@@ -97,15 +94,17 @@ class ModelTypedIRAddress {
   ModelTypedIRAddress &operator=(ModelTypedIRAddress &) = default;
 
 public:
-  ModelTypedIRAddress(const model::QualifiedType &PT, Value *A) :
-    PointeeType(PT), Address(A) {
-    revng_assert(PointeeType.verify() and nullptr != Address);
+  ModelTypedIRAddress(model::UpcastableType &&Pointee, Value *A) :
+    PointeeType(std::move(Pointee)), Address(A) {
+
+    isValid();
```

Comment:
???

Reply:
It was weird for it to have two distinct "invalid" states, so I merged them into one since it otherwise only affects the `dump` helper.

Reply:
1. Does `isValid` have side effects? `isXXX` methods should be `const`.
2. If it doesn't have side effects, why are you invoking it ignoring its return value?

Reply:
Done.